tests/integration/query_helpers/test_query_docs.py

```diff
@@ -25,17 +23,11 @@ def setup_faiss():
     ]
     embeddings = HuggingFaceEmbeddings(model_name="all-MiniLM-L6-v2")
 
-    file_path = "./index/faiss_index"
-
-    if os.path.exists(file_path):
```

Comment:
I'm not sure about that. The difference is just if it is initialized from a file or variable.

Reply:
If the vector index is already present load the embeddings from the index or create the index and store the embeddings. You do not want to create the index for every run or every tests that you run. These tests will increase over period of time, and if we have to create the index and store the embeddings for every run, it will make the tests extremely slow.

Reply:
That is exactly how it is implemented now. If the original idea was to create index store just once in tests, the current implementation is not matching it. Each test initializes a new index store either from file or from the variable (the fixture is not persisting between different tests).

These tests (`test_query_docs.py`) are currently the slowest integration tests we have, because of it.

I can't say if it is a good idea to reuse the same index store in multiple tests - maybe if we ensure they are not mutating it during testing it would be ok.

And from my testing, it doesn't matter if the store is initialized from file or variable. The execution time is the same. It means that most of the execution time is not coming from loading the data, but from setting up the index. Maybe I have something special on my computer, so you can test it too.

So one thing is to say we want to reuse the store between tests.
The other thing is how to populate it. If there is no difference in execution time, IMHO it would be better to explicitly fill it in code (as is in `list_of_documents`) - unless the index would need to have a lot of complicated documents (which doesn't sound right for unit testing though).

Reply:
What do you mean the current implementation does not match it? It is been set up in a fixture and that fixture is used by the tests. So if the index is already there it will not create it again. You will see the difference in test run execution time, once we have more docs in the vector store. ```And from my testing, it doesn't matter if the store is initialized from file or variable```.  Where do you even see that vector store is created from variable? I guess you are  running into issue for this test on your local machine.  BTW these are integration tests, think of it a in memory db test so it will be slower than the other test. So, if you find this to be slow, probably we can say that we need to have proper db and not file based system for vector db as been investigated here :) and avoid doing this JIRA https://issues.redhat.com/browse/OLS-120 cc @bparees

Reply:
BTW, these tests are been used by whole of langchain community

Reply:
@bparees Setting up the index store every time will mean creating embeddings for the docs every time which will burn up the cycles of the testing.

Reply:
> It is been set up in a fixture and that fixture is used by the tests.

Every test that uses that fixture executes it - its not "cached" between tests
So in each test it is doing either `db = FAISS.load_local(file_path, embeddings)` or `db = FAISS.from_documents(list_of_documents, embeddings)` and then storing db again after each test is executed.

> once we have more docs in the vector store

IMHO this is a bad pattern for tests. Instead of creating a huge index with all files that are needed for various tests, you should have an independent index (with relevant documents) for each test. If creating an index is an expensive operation, you can reuse the db, but you need to flush it and populate with documents (relevant for the test) for each test.

> Where do you even see that vector store is created from variable?

https://github.com/openshift/lightspeed-service/blob/main/tests/integration/query_helpers/test_query_docs.py#L33

I think you've implemented the switch to "speed up" the tests. Create an index in the first run, store it locally and reuse it later. But it seems the fundament on which you've based your decision is wrong. For such a small "content" of the index/db, it doesn't matter if you load it from file or shove it directly there on fly. You are not saving any time.

And the [switch](https://github.com/openshift/lightspeed-service/blob/main/tests/integration/query_helpers/test_query_docs.py#L28) is not checking the existence of store index, but just the dir where it should live. That is why run into this issue in the first place.